src/test/java/org/agbrothers/englishbot/service/LessonServiceTest.java

```diff
@@ -10,96 +12,77 @@
 import static org.mockito.Mockito.*;
 
 class LessonServiceTest {
+    final LessonRepository lessonRepository = mock(LessonRepository.class);
     final WordPoolService wordPoolService = mock(WordPoolService.class);
-    final LessonService lessonService = new LessonService(wordPoolService);
+    final LessonService lessonService = new LessonService(wordPoolService, lessonRepository);
 
     @Test
     void getLesson_LessonWithCurrentWordExists_ExpectNoLessonCreation() {
         List<Word> words = Collections.singletonList(new Word());
-        Lesson lesson = new Lesson(words, words);
+        User user = new User();
+        Lesson lesson = new Lesson(words, words, user);
         lesson.setCurrentWord(new Word());
-        String chatId = "testChatId";
-        Map<String, Lesson> lessonRegistry = new HashMap<>();
-        lessonRegistry.put(chatId, lesson);
-        lessonService.setLessonRegistry(lessonRegistry);
 
-        Lesson result = lessonService.getLesson(chatId);
+        when(lessonRepository.findLessonByUser(user)).thenReturn(lesson);
+
+        lessonService.getLesson(user);
 
-        assertEquals(lesson, result);
         verify(wordPoolService, never()).getRandomWordPool();
```

Comment:
in test signature, it is said `ExpectNoLessonCreation` but you check for not calling `getRandomWordPool`
please, add needed checks

Reply:
Thanks for comment.